agent/php_internal_instrument.c

```diff
@@ -849,6 +849,33 @@ NR_INNER_WRAPPER(mysqli_real_connect) {
   zval* mysqli_obj = NULL;
   int zcaught = 0;
 
+  /* PHP 8.1 and later will report a deprecation warning if null is sent where
+   * a non-null argument value is expected.  For these PHP versions we use the
+   * same argument type specification as the mysqli::real_connect() extension
+   * uses to avoid creating this deprecation warning.
+   * For older PHPs continue to use the same specification string as previously
+   * to minimize any chances of introducing new problems.
+   */
+#if ZEND_MODULE_API_NO >= ZEND_8_1_X_API_NO
+  if (FAILURE
+      == zend_parse_parameters_ex(
+          ZEND_PARSE_PARAMS_QUIET, ZEND_NUM_ARGS() TSRMLS_CC, "O|s!s!s!s!l!s!l",
```

Comment:
```suggestion
          ZEND_PARSE_PARAMS_QUIET, ZEND_NUM_ARGS() TSRMLS_CC, "o|s!s!s!s!l!s!l",
```
I think this is the cause of the test failures for 8.1 and 8.2.
According to the zpp api [doc](https://github.com/php/php-src/blob/master/docs/parameter-parsing-api.md),
- `O` is for an object of specific type given by class entry
- `o` is for a generic object (typical `zval *`)

Reply:
It's also the primary difference between this and the previous logic, which used `o`. I tested this locally and it fixed the issue.